acstools/acszpt.py

```diff
@@ -0,0 +1,379 @@
+
+            'SBC': ['F115LP', 'F122M', 'F125LP',
+                    'F140LP', 'F150LP', 'F165LP']
+        }
+        self._zpt_table = None
+
+        # Set the private attributes
+        if filt is None:
+            self._url = ('https://acszeropoints.stsci.edu/results_all/?date={}'
+                         '&detector={}'.format(self.date, self.detector))
+        else:
+            self._filt = filt.upper()
+            self._url = ('https://acszeropoints.stsci.edu/'
+                         'results_single/?date1={0}&detector={1}'
+                         '&{1}_filter={2}'.format(self.date,
+                                                  self.detector,
+                                                  self.filt))
+        # ACS Launch Date
+        self._acs_installation_date = dt.datetime(2002, 3, 7)
+        # The farthest date in future that the component and throughput files
+        # are valid for. If input date is larger, extrapolation is not valid.
+        self._extrapolation_date = dt.datetime(2021, 12, 31)
+        self._msg_div = '-' * 79
+        self._valid_detectors = ['HRC', 'SBC', 'WFC']
+        self._response = None
+        self._failed = False
+        self._data_units = {
+            'FILTER': u.dimensionless_unscaled,
+            'PHOTPLAM': u.angstrom,
+            'PHOTFLAM': u.erg / u.cm ** 2 / u.second / u.angstrom,
+            'STmag': u.STmag,
+            'VEGAmag': u.mag,
+            'ABmag': u.ABmag
+        }
+        self._block_size = len(self._data_units)
+
+    @property
+    def date(self):
+        """The user supplied date. (str)"""
+        return self._date
+
+    @property
+    def detector(self):
+        """The user supplied detector. (str)"""
+        return self._detector
+
+    @property
+    def filt(self):
+        """The user supplied filter, if one was given. (str or `None`)"""
+        return self._filt
+
+    @property
+    def zpt_table(self):
+        """The results returned by the ACS Zeropoint Calculator. (`astropy.table.QTable`)"""
+        return self._zpt_table
+
+    def _check_inputs(self):
+        """Check the inputs to ensure they are valid.
+
+        Returns
+        -------
+        status : bool
+            True if all inputs are valid, False if one is not.
+
+        """
+        valid_detector = True
+        valid_filter = True
+        valid_date = True
+        # Determine the submitted detector is valid
+        if self.detector not in self._valid_detectors:
+            msg = ('{} is not a valid detector option.\n'
+                   'Please choose one of the following:\n{}\n'
+                   '{}'.format(self.detector,
+                               '\n'.join(self._valid_detectors),
+                               self._msg_div))
+            LOG.error(msg)
+            valid_detector = False
+
+        # Determine if the submitted filter is valid
+        if (self.filt is not None and
```

Comment:
You need to add back the `valid_detector` check here, otherwise you will get `KeyError`. I took that out in my refactoring, which you reverted to.

Reply:
Good catch!